grpc/src/main/java/com/linecorp/armeria/server/grpc/StreamingServerCall.java

```diff
@@ -265,6 +265,6 @@ public void transportReportStatus(Status status, Metadata unused) {
             // failure there's no need to notify the server listener of it).
             return;
         }
-        closeListener(status, false, true);
+        closeListener(status, metadata, false, true);
```

Comment:
> // "A server doesn't see trailers from the client so will never have Metadata here." (on top of this method)

Note that I checked client doesn't send metadata even if `onError(Throwable)` is called in client stream. (It cancels stream with `Status.CANCEL` without propagating error details.)
But couldn't find the code that server ignores trailers. So I'm not confident here.

Reply:
As per gRPC HTTP2 wire format. a client does not send trailers
https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests

I don't understand the context of the comment. Armeria gRPC runtime implementation has changed a lot. So the comment could be obsolete. Anyway, the code looks okay. Only non-empty metadata will be injected when an error is raised while deserialization. 

Side note: `TransportStatusListener` has existed when gRPC runtime used a custom stream process instead of Reactive Streams. #2981 At that time, `TransportStatusListener` was useful to propagate errors that happened amid stream processing. But now, we can deliver the errors via `Subscriber.onError()` or `StreamDecoder.processOnError()` without `TransportStatusListener`